pkg/sql/conn_executor_exec.go

```diff
@@ -988,6 +989,12 @@ func (ex *connExecutor) commitSQLTransactionInternal(ctx context.Context) error
 	ctx, sp := tracing.EnsureChildSpan(ctx, ex.server.cfg.AmbientCtx.Tracer, "commit sql txn")
 	defer sp.Finish()
 
+	for n, c := range ex.extraTxnState.sqlCursors.list() {
```

Comment:
I think it would be better to put this logic nearby where we close all of the cursors, perhaps even inside of `closeAll`. You might need to pass a boolean to `closeAll` because we also use it on server shutdown, but I feel this would prevent future refactorings from "forgetting" to do this check.

Reply:
also need one for `rollback` too right?

Reply:
yeah, we'll need to make sure `txnRollback` and `txnRestart` transitions are handled too, but that seems fine